### PR TITLE
py-httpstan: Add new package

### PR DESCRIPTION
**Pull request by haralmha**

### CODE REVIEW

**Comment by haralmha** [comment]
Depends on py-marshmallow and py-webargs, which are not yet added to spack. PRs made here: https://github.com/spack/spack/pull/29764 and here: https://github.com/spack/spack/pull/29768

File: var/spack/repos/builtin/packages/py-httpstan/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    """HTTP-based REST interface to Stan, a package for Bayesian inference."""
+
+    homepage = "https://mc-stan.org/"
+    url      = "https://github.com/stan-dev/httpstan/archive/refs/tags/4.6.1.tar.gz"
+
+    maintainers = ['haralmha']
+
+    version('4.6.1', sha256='703e5e04e60651e0004574bb9695827d759fd13eb0d6bd67f827c1bfa0a1fd31')
+
+    depends_on('python@3.7:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httpstan/package.py]:
```suggestion
    depends_on('python@3.7:3', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-httpstan/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on('py-setuptools@41.0:', type='build')
+    depends_on('py-poetry-core@1.0.0:', type='build')
+    depends_on('py-aiohttp@3.7:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httpstan/package.py]:
```suggestion
    depends_on('py-aiohttp@3.7:3', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-httpstan/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on('py-appdirs@1.4:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httpstan/package.py]:
```suggestion
    depends_on('py-appdirs@1.4:1', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-httpstan/package.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on('py-webargs@8.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httpstan/package.py]:
```suggestion
    depends_on('py-webargs@8.0:8', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-httpstan/package.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on('py-marshmallow@3.10:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httpstan/package.py]:
```suggestion
    depends_on('py-marshmallow@3.10:3', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-httpstan/package.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on('py-numpy@1.16:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httpstan/package.py]:
```suggestion
    depends_on('py-numpy@1.16:1', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Caret deps work a bit differently: https://python-poetry.org/docs/dependency-specification/

File: var/spack/repos/builtin/packages/py-httpstan/package.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+    homepage = "https://mc-stan.org/"
+    url      = "https://github.com/stan-dev/httpstan/archive/refs/tags/4.6.1.tar.gz"
+
+    maintainers = ['haralmha']
+
+    version('4.6.1', sha256='703e5e04e60651e0004574bb9695827d759fd13eb0d6bd67f827c1bfa0a1fd31')
+
+    depends_on('python@3.7:3', type=('build', 'run'))
+    depends_on('py-setuptools@41.0:', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httpstan/package.py]:
```suggestion
    depends_on('py-setuptools@41.0:', type=('build', 'run'))
```
`setuptools` is imported at run-time

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Comment by alalazo** [comment]
@haralmha I would advise submitting a unique PR with related software. That would make package audits pass the first time, since there will be no missing versions or missing dependencies. With that said: thanks a lot for your contributions!

**Comment by haralmha** [comment]
@alalazo ah okay, thanks for your feedback! I was not sure if you wanted one PR per package or one for the whole bunch, but I'll go with the latter next time :)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**